pkg/nsx/services/subnet/subnet.go

```diff
@@ -194,6 +197,30 @@ func (service *SubnetService) ListSubnetSetID(ctx context.Context) sets.Set[stri
 	return subnetsetIDs
 }
 
+func (service *SubnetService) ListSubnetByName(ns, name string) []*model.VpcSubnet {
+	nsxSubnets := service.SubnetStore.GetByIndex(common.TagScopeNamespace, ns)
```

Comment:
Shall we add an index function for TagScopeNamespace?

Reply:
Thanks for reviewing. Updated.
Add two index functions, `common.TagScopeVMNamespace:    subnetIndexNamespaceFunc,` for Subnet and SubnetSet.
And `common.TagScopeNamespace:      subnetIndexScopeNamespaceFunc` for default Pod SubnetSet.